### PR TITLE
Add empty value check for hazelcast.licensekey overriding config

### LINKED ISSUE #19849: Unrelated Log Messages when empty License Key is set
**Describe the bug**
When Hazelcast License Key is set to empty string, the error log does not return an understandable message. This is relevant because this is what happens when I use hazelcast-enterprise Helm Chart without setting the field `hazelcast.licenseKey`.
image

**Expected behavior**
It should return something similar to "com.hazelcast.license.exception.InvalidLicenseException: Invalid License Key!"

**To Reproduce**

1. Set "HZ_LICENSEKEY" to empty string
2. Run the Hazelcast member

**Additional context**

### PR DESCRIPTION
**Pull request by hasancelik**
With this check, `java.lang.IllegalArgumentException: License key can not be empty.` will be thrown instead of `StringIndexOutOfBoundsException`

fixes #19849

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public ClientConfig overwriteClientConfig(ClientConfig config) {
                   properties.entrySet().stream()
                     .filter(e -> !unprocessed.containsKey(e.getKey()))
                     .map(e -> {
-                        if (e.getKey().equals("hazelcast.licensekey")) {
+                        if (e.getKey().equals("hazelcast.licensekey") && !e.getValue().isEmpty()) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java:86]: Can we make this more resilient to invalid input? The code below will still throw `StringIndexOutOfBoundsException` if input license key is `asdf`. It seems `originalKeyPart` needs to have length at least 5 for the `substring`s to work.

Comment (ldziedziul) [reply hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java:86]: Would be great to provide unit tests for empty/null/short license keys

**Review by ldziedziul — APPROVED** [review]
Looks good 👍

**Review by vbekiaris — APPROVED** [review]
@hasancelik thanks for the fix!

**Pull request merged**